src/test/java/training/sign_up/SignUp.java

```diff
@@ -0,0 +1,39 @@
+package training.sign_up;
+
+import com.intuit.karate.junit5.Karate;
+import com.intuit.karate.Results;
+import com.intuit.karate.Runner;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import net.masterthought.cucumber.Configuration;
+import net.masterthought.cucumber.ReportBuilder;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+public class SignUp {
+
+    @Karate.Test
+    Karate testUsers() {
+        return Karate.run("sign_up").relativeTo(getClass());
+    }
+
+    private static void generateReport(String karateOutputPath) {
+        Collection<File> jsonFiles = FileUtils.listFiles(new File(karateOutputPath), new String[]{"json"}, true);
+        List<String> jsonPaths = new ArrayList<>(jsonFiles.size());
+        jsonFiles.forEach(file -> jsonPaths.add(file.getAbsolutePath()));
+        Configuration config = new Configuration(new File("target"), "Name-API");
+        ReportBuilder reportBuilder = new ReportBuilder(jsonPaths, config);
+        reportBuilder.generateReports();
+    }
+
+    @Test
+    public void testParallel() {
+        String karateOutputPath = "target/surefire-reports";
+        Results results = Runner.parallel(getClass(), 5, karateOutputPath);
+        generateReport(results.getReportDir());
+        Assertions.assertEquals(0, results.getFailCount());
+    }
```

Comment:
don't generate a report in every runner that you create. This code and this report must be in a general runner which execute all the features (in the future)